power_grid_model_c/power_grid_model/include/power_grid_model/common/three_phase_tensor.hpp

```diff
@@ -284,6 +284,34 @@ inline std::pair<DoubleComplex, DoubleComplex> inv_sym_param(DoubleComplex const
     return {(s + m) * det_1, -m * det_1};
 }
 
+// inverse of symmetric tensor
+template <symmetry_tag sym> inline ComplexTensor<sym> inv_sym_tensor(ComplexTensor<sym> const& x) {
+    if constexpr (is_symmetric_v<sym>) {
+        return 1.0 / x;
+    } else {
+        auto [s, m] = inv_sym_param(x(0), x(1));
+        return ComplexTensor<sym>(s, m);
+    }
+}
+
+// inverse of Y via s, m
+template <symmetry_tag sym> inline ComplexTensor<sym> inv_y(ComplexTensor<sym> const& y) {
```

Comment:
`y` is very specific EE terminology. This file is about the generic mathematical concept of a 3x3 tensor together with the conventions about the basis vectors (`1,a,a²`) that stems from EE, but is actually more generic.

The explicit reference to `y` does not belong here, or requires different terminology (e.g. referencing a more generic symmetric tensor `t`).

See e.g. how we do not explicitly access `(0)` and `(1)` when calling `inv_sym_param` above this. Instead, the input is diagonal part `s` and off-diagonal part `m`. As such, usage is e.g. `inv_sym_param(y(0), y(1))`.

There are 2 options to solve it here:
1. change the input type from `ComplexTensor<sym>` to a `ComplexSymmetricTensor<sym>`, which needs to be created because it doesn't exist yet. NOTE: this also requires defining the related operators as we did for `DiagonalTensor<sym>`. `Y` would have to be redefined in multiple locations. While I think that this is worth the effort (less, more stable matrix calculations), it's also very much a scope-creep and I don't think we should go down this route in this PR.
2. use the same formalism as for `inv_sym_param` here, with `DoubleComplex const& s, DoubleComplex const& m` as input parameters

Reply:
Due to the rework being done to try to remove instabilities in a different way, this shall be removed as it is no longer needed at the moment. However, I believe your suggestion in 1 is very valid and may be worth investigating if it is of value outside of this PR, and if it is, we should create an issue with it.

Reply:
After going to 012 domain in the asymmetric case, there is no longer need for such an inverse. This has been removed.